google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/GrpcResultSetTest.java

```diff
@@ -665,4 +665,161 @@ private void verifySerialization(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(Type.StructField.of("f", Type.int64()))))
+            .addValues(Value.int64(Long.MIN_VALUE).toProto())
+            .addValues(Value.int64(Long.MAX_VALUE).toProto())
+            .build());
+    consumer.onCompleted();
+
+    resultSet.next();
+    Long l = resultSet.getLong(0);
+    assertThat(l.equals(Long.MIN_VALUE));
+
+    resultSet.next();
+    l = resultSet.getLong(0);
+    assertThat(l.equals(Long.MAX_VALUE));
+  }
+
+  @Test
+  public void getDate() {
+    consumer.onPartialResultSet(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(Type.StructField.of("f", Type.date()))))
+            .addValues(Value.date(Date.fromYearMonthDay(2018, 5, 29)).toProto())
+            .build());
+    consumer.onCompleted();
+
+    resultSet.next();
+    assertThat(resultSet.getDate(0).compareTo(Date.fromYearMonthDay(2018, 5, 29)) == 0);
+  }
+
+  @Test
+  public void getTimestamp() {
+    consumer.onPartialResultSet(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(Type.StructField.of("f", Type.timestamp()))))
+            .addValues(Value.timestamp(Timestamp.parseTimestamp("0001-01-01T00:00:00Z")).toProto())
+            .build());
+    consumer.onCompleted();
+
+    resultSet.next();
+    assertThat(resultSet.getTimestamp(0) == Timestamp.parseTimestamp("0001-01-01T00:00:00Z"));
+  }
+
+  @Test
+  public void getBooleanArray() {
+    boolean[] bArray = {true, true, false};
+    consumer.onPartialResultSet(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(Type.StructField.of("f", Type.array(Type.bool())))))
+            .addValues(Value.boolArray(bArray).toProto())
+            .build());
+    consumer.onCompleted();
+    resultSet.next();
+    assertThat(resultSet.getBooleanArray(0)).isEqualTo(bArray);
+  }
+
+  @Test
+  public void getLongArray() {
+    long[] lArray = {111, 333, 444, 0, -1, -2234, Long.MAX_VALUE, Long.MIN_VALUE};
+
+    consumer.onPartialResultSet(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(Type.StructField.of("f", Type.array(Type.int64())))))
+            .addValues(Value.int64Array(lArray).toProto())
+            .build());
+    consumer.onCompleted();
+    resultSet.next();
+    assertThat(resultSet.getLongArray(0)).isEqualTo(lArray);
+  }
+
+  @Test
+  public void getDoubleArray() {
+    double[] dArray = {Double.MAX_VALUE, Double.MIN_VALUE, 111, 333, 444, 0, -1, -2234};
+
+    consumer.onPartialResultSet(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(
+                Type.StructField.of("f", Type.array(Type.float64())))))
+            .addValues(Value.float64Array(dArray).toProto())
+            .build());
+    consumer.onCompleted();
+    resultSet.next();
+
+    assertThat(resultSet.getDoubleArray(0).equals(dArray));
```

Comment:
For this and below, `assertThat(resultSet.getXXX(0)).isEqualTo(YYY)`.

Reply:
Hm, I tried to do this but Intellij showed it as struck out i.e. deprecated. Is this allowed for double array?